mio 0.6.17 compilation error on apple ios targets

Mio version 0.6.17 fails to compile on `apple-ios-i386` and `armv7s-apple-ios` with the following error:

```
error[E0277]: the trait bound `i32: std::convert::From<u32>` is not satisfied
  --> /Users/jenkins/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.6.17/src/sys/unix/kqueue.rs:77:26
   |
77 |                 tv_nsec: libc::c_long::from(to.subsec_nanos()),
   |                          ^^^^^^^^^^^^^^^^^^ the trait `std::convert::From<u32>` is not implemented for `i32`
   |
   = help: the following implementations were found:
             <i32 as std::convert::From<bool>>
             <i32 as std::convert::From<i16>>
             <i32 as std::convert::From<i8>>
             <i32 as std::convert::From<std::num::NonZeroI32>>
           and 2 others
   = note: required by `std::convert::From::from`
```

--- a/src/sys/unix/kqueue.rs
+++ b/src/sys/unix/kqueue.rs
@@ -74,7 +74,11 @@ impl Selector {
         let timeout = timeout.map(|to| {
             libc::timespec {
                 tv_sec: cmp::min(to.as_secs(), time_t::max_value() as u64) as time_t,
-                tv_nsec: libc::c_long::from(to.subsec_nanos()),
+                // `Duration::subsec_nanos` is guaranteed to be less than one
+                // billion (the number of nanoseconds in a second), making the
+                // cast to i32 safe. The cast itself is needed for platforms
+                // where C's long is only 32 bits.
+                tv_nsec: libc::c_long::from(to.subsec_nanos() as i32),
             }
         });
         let timeout = timeout.as_ref().map(|s| s as *const _).unwrap_or(ptr::null_mut());

